query.go

```diff
@@ -0,0 +1,114 @@
+package ensign
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	api "github.com/rotationalio/go-ensign/api/v1beta1"
+)
+
+// QueryCursor exposes event results from an EnSQL query with familiar database cursor
+// semantics.
+type QueryCursor struct {
+	stream api.Ensign_EnSQLClient
+}
+
+// read the next event from the stream and return it to the caller.
+func (c *QueryCursor) read() (event *Event, err error) {
+	if c.stream == nil {
+		return nil, ErrCursorClosed
+	}
+
+	var wrapper *api.EventWrapper
+	if wrapper, err = c.stream.Recv(); err != nil {
+		// Return nil if the stream is closed
+		if err == io.EOF {
+			return nil, nil
```

Comment:
Should we do a CloseSend here to close the client side of the stream? I'm actually not sure about the gRPC server-side streaming semantics, so it would be worth looking up. At the very least, it might be good to set the stream to nil to prevent future reads?